rust/opendp/src/sarus/privacy_loss_distribution.rs

```diff
@@ -0,0 +1,212 @@
+                ((Rational::from(1)-&last_delta)*&exp_epsilon-(Rational::from(1)-&delta)*&last_exp_epsilon)/&denom,
+            ));
+            last_exp_epsilon = exp_epsilon.clone();
+            last_delta = delta.clone();
+        }
+        // Compute probabilities
+        let mut last_alpha = result_alpha_betas[0].0.clone();
+        let mut last_beta= result_alpha_betas[0].1.clone();
+        for i in 1..result_alpha_betas.len() {
+            let alpha = result_alpha_betas[i].0.clone();
+            let beta= result_alpha_betas[i].1.clone();
+            if &beta != &last_beta {
+                let exp_epsilon = -(alpha.clone()-&last_alpha)/(beta.clone()-&last_beta);
+                result_exp_privacy_loss_probabilities.push((exp_epsilon, -(beta.clone()-&last_beta)));
+            }
+            last_alpha = alpha.clone();
+            last_beta = beta.clone();
+        }
+        PLDistribution::new(result_exp_privacy_loss_probabilities)
+    }
+}
+
+/// Compute the composition of PLDs
+impl Mul for &PLDistribution {
+    type Output = PLDistribution;
+    fn mul(self, other: &PLDistribution) -> PLDistribution {
+        let mut result = PLDistribution {exp_privacy_loss_probabilities:BTreeMap::new()};
+        for (s_p_y_x,s_p_x) in &self.exp_privacy_loss_probabilities {
+            for (o_p_y_x,o_p_x) in &other.exp_privacy_loss_probabilities {
+                // s,o
+                let so_p_y_x = s_p_y_x.clone() * o_p_y_x;
+                let so_p_x = s_p_x.clone() * o_p_x;
+                result.exp_privacy_loss_probabilities.entry(so_p_y_x)
+                    .and_modify(|p_x| { *p_x += so_p_x.clone() })
+                    .or_insert(so_p_x);
+                // s_inv,o
+                if s_p_y_x>&Rational::from(0) && s_p_y_x<&Rational::from(1) {
+                    let so_p_y_x = s_p_y_x.clone().recip() * o_p_y_x;
+                    let so_p_x = s_p_x.clone()*s_p_y_x * o_p_x;
+                    if so_p_y_x<=Rational::from(1) {
+                        result.exp_privacy_loss_probabilities.entry(so_p_y_x)
+                            .and_modify(|p_x| { *p_x += so_p_x.clone() })
+                            .or_insert(so_p_x);
+                    }
+                }
+                // s,o_inv
+                if o_p_y_x>&Rational::from(0) && o_p_y_x<&Rational::from(1) {
+                    let so_p_y_x = o_p_y_x.clone().recip() * s_p_y_x;
+                    let so_p_x = s_p_x.clone() * o_p_x * o_p_y_x;
+                    if so_p_y_x<=Rational::from(1) {
+                        result.exp_privacy_loss_probabilities.entry(so_p_y_x)
+                            .and_modify(|p_x| { *p_x += so_p_x.clone() })
+                            .or_insert(so_p_x);
+                    }
+                }
+            }
+        }
+        result
+    }
+}
+
+impl Default for PLDistribution {
+    fn default() -> Self {
+        PLDistribution::new([(Rational::from(1),Rational::from(1))])
```

Comment:
Minor, but this PR won't compile unless I tweak this into an iterator:
```suggestion
        PLDistribution::new(vec![(Rational::from(1), Rational::from(1))].into_iter())
```